.github/workflows/tests.yml

```diff
@@ -28,20 +54,70 @@ jobs:
           key: ${{ matrix.job.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ matrix.job.os }}-m2
 
-      - name: Compile and install everything
-        run: mvn clean install -DskipTests
-
       - name: Run Andy unit tests
         run: mvn test -pl andy -DexcludedGroups=assignments
 
+  run_assignment_tests:
+    needs: checkstyle
+    runs-on: ${{ matrix.job.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          - os: ubuntu-22.04
+          - os: macos-12
```

Comment:
Should we also run tests on Windows? #143 is fixed now.

Reply:
Let's do that in another PR too!